packages/aic/src/document.ts

```diff
@@ -0,0 +1,318 @@
+import { sign, validateSig } from './signature'
+import { pid } from './pid'
+import { ErrorMessage, Value, TidString, DidKeyString, Patch, Diff, Diffs, Document, Tick, Asymmetric } from './types'
+
+export { pid, sign, validateSig }
+
+/*
+  Example identity tick
+  {
+    "tid": "3j5r-ts5-ojpm-2c",
+    "did": "did:aic:zrr5lxhs4rjlowv5",
+    "diffs": {
+      "3j5r-ts5-o7x4-2c": {
+        "adx/account_keys": [ "did:key:zDnaeycJUNQugcrag1WmLePtK9agLYLyXvscnQM4FHm1ASiRV" ],
+        "adx/recovery_keys": [ "did:key:zDnaeYHbbWiaCwAXvHXRyVqENDv8Sr3fwHW8P5eakkz4MqsTa" ]
+      },
+      "3j5r-ts5-ocuu-2c": {
+        "prev": "3j5r-ts5-o7x4-2c",
+        "patches": [ [ "put", [ "name" ], "aaron.blueskyweb.xyz" ] ],
+        "key": "did:key:zDnaeycJUNQugcrag1WmLePtK9agLYLyXvscnQM4FHm1ASiRV",
+        "sig": "z4PWrudF88YEBqPQPhfWpiL7MFqhHhCWuKn8Ri4MyKZRcRLbSMFgP3WYKTBsfnq5TXGR9wbjJKKPSb7i1kQgisHzf"
+      },
+      "3j5r-ts5-ofsm-2c": {
+        "prev": "3j5r-ts5-ocuu-2c",
+        "patches": [ [ "del", [ "adx/account_keys" ] ] ],
+        "key": "did:key:zDnaeycJUNQugcrag1WmLePtK9agLYLyXvscnQM4FHm1ASiRV",
+        "sig": "zSFwunTYyAh5ePkDgJ1Gw1nCtnqpVs1QzfXzkB9bHvoLoYKNeKsfeFHvToKT6tLiB5y9H5CBH1jgeYPtK57kodh7"
+      },
+      "3j5r-ts5-ojpm-2c": {
+        "prev": "3j5r-ts5-ofsm-2c",
+        "patches": [ [ "put", [ "hello" ], "world" ] ],
+        "key": "did:key:zDnaeycJUNQugcrag1WmLePtK9agLYLyXvscnQM4FHm1ASiRV",
+        "sig": "z431h9jjokS7TJ3VWT6XtbSUfTFVGUrLQi2uNMU4PbRXSNLoZS13dkkyeBrnfRrHdpiZhYGx6LCYSw5GgdEiDfEm9"
+      }
+    },
+    "key": "did:key:zDnaeeL44gSLMViH9khhTbngNd9r72MhUPo4WKPeSfB8xiDTh",
+    "sig": "z3naShLa1KV5ZV5raHbuAAC2zjjyBbTp4htmCA42GKVLDQi7kSnFCteg7n5ap5uJMpsjdxLgL8E6kRR9cqyt6zqK6"
+  }
+makes
+  {
+    'adx/recovery_keys': [ 'did:key:zDnaeYHbbWiaCwAXvHXRyVqENDv8Sr3fwHW8P5eakkz4MqsTa'],
+    id: 'did:aic:zrr5lxhs4rjlowv5',
+    name: 'aaron.blueskyweb.xyz'
+  }
+*/
+
+export const updateTick = async (
+  did: string, // the did:aic: being updated
+  tid: TidString, // the consenus time of the tick's generation
+  candidateDiff: Diff | Document | null, // null when only updating tid
+  stored_tick: Tick | null, // null when there is not db entry (did init)
+  key: Asymmetric, // the consortium passes in the key
+): Promise<Tick | ErrorMessage> => {
+  // @TODO validate that tid > all tids in the diffs
+  if (stored_tick === null) {
+    // this is the  initial registration of a did:aic
+    return registerNewDid(did, tid, candidateDiff, key)    
+  }
+
+  // since stored_tick is not null this is an update
+  // of a registered did:aic
+  if (!(await validateSig(stored_tick, key))) {
+    // this means the consortium data base has been corupted
+    return { error: 'bad signature: stored tick' }
+  }
+  if (stored_tick.key != key.did()) {
```

Comment:
kind of a nit cause I don't think it matters here, but I'd get in the habit of always doing `!==` for equality checks.